irohad/network/impl/peer_communication_service_impl.cpp

```diff
@@ -38,6 +41,12 @@ namespace iroha {
       return ordering_gate_->on_proposal();
     }
 
+    rxcpp::observable<
+        std::shared_ptr<iroha::validation::VerifiedProposalAndErrors>>
```

Comment:
Can't it be without shared_ptr wrapper?

Reply:
The initial implementation was in raw ```VerifiedProposalAndErrors```, but then I switched to shared_ptr, because there is possibly large vector, and it gets transferred much a lot between the code